src/common.js

```diff
@@ -450,6 +451,7 @@ var DEFAULT_WRAP_OPTIONS = {
   canReceivePipe: false,
   cmdOptions: null,
   globStart: 1,
+  handlesFatalDynamically: false,
```

Comment:
I need another look to wrap my head around this option. Not necessarily opposed to this, though

Reply:
Ok, I see what's going on. This approach is fine, although I think the nicer long-term solution would be one of:

1. Permit an `options` parameter for all commands, and allow this to temporarily override `shell.config` (the next command will get the original copy of `shell.config`). Ex. `shell.cp('foo', 'bar', { fatal: true })`. I implemented this sort of parsing in #387, but for a different reason.
2. Reset `shell.config` after each command exits, unless it declares it mutates `shell.config` (we would need this option to support `shell.set('-e')`).

I was originally thinking this would look more like option 2 (set `shell.config.fatal = opts.fatal;` and let it get reset when the command exits), but that would require a bit more work to support `shell.set()`.

Reply:
Actually, for either of these we would need to make sure a snapshot of `shell.config` gets passed down to the closure returned by `wrap()`, to avoid race conditions if it's modified in the middle of an async method. But that seems feasible.

Reply:
Agreed. Long term I think there's a better way to approach this problem that's more intuitive and reusable, but it'd probably require refactoring a fair bit of the existing functionality. Something to think about for the future!